test/test_sparse.py

```diff
@@ -2177,6 +2177,27 @@ def test_factory_empty_indices(self):
         expected_indices = torch.empty((4, 0), dtype=torch.long, device=device)
         self.assertEqual(tensor._indices(), expected_indices)
 
+    def test_sparse_abs(self):
+        device = 'cuda' if self.is_cuda else 'cpu'
+
+        if device == 'cpu':
```

Comment:
`torch.tensor([....], dtype=torch.long, device=device)`, similar for values, that way you don't need `device` conditional.

Reply:
will update the test case as per this